### ISSUE TITLE
Min vals division by zero in lazyrepeatarrays may be causing overflow errors

### ISSUE THREAD

**Issue by IshanMi** [OpenMined/PySyft#6950]
When we compute the final min_vals and max_vals, we often do something like: min_min, min_max, max_max, max_min and compare all the values to see which is the smallest.

For an op like truediv this might involuntarily involve dividing by zero and raising an error.

co-authored by: @teo-milea

**Comment by IshanMi** [comment]
![Image](https://user-images.githubusercontent.com/32711264/199743372-c9bb595e-8510-426f-84b8-3ef7b82c6186.png)

**Comment by IshanMi** [comment]
This led to this in the logs:



![Image](https://user-images.githubusercontent.com/32711264/199743504-8118bf83-be22-424d-ad1b-c6dcdc56d976.png)

**Comment by iamtrask** [comment]
As a bug this is more complex than a simple overflow issue. It is genuinely an infinite privacy spend risk WRT the public privacy API. If the denominator has a min_val of -1 and a max_val of 1, then it's possible that the private value is indeed zero and the computation is invalid. The permutation of min_min, min_mmax, etc. is correctly capturing this. That is to say - it's correct to determine that it's possible this is an invalid computation.

We will need to spend more time thinking on this. For 0.7 - please fix by raising an exception on the pointer when a denominator could be zero (isn't positive-definite or negative-definite). The exception should read something like, "I'm sorry, but our privacy accountant can't yet divide by a number which might be zero. Please find a way to approximate your computation without dividing by a private value which might be zero. This can usually be done by computing the function piecewise - by performing three computations which are merged through masking and summing: one which addresses what happens if the denominator is positive-definite, another which addresses what happens if the denominator is negative-definite, and another which addresses what happens if the denominator is 0 exactly. Use comparison operators, masking, and summing to accomplish this (not if statements - which don't work on private values). "

**Comment by teo-milea** [comment]
In order to be more general, during the publishing system I raised an exception when we find any `NaN` or `Inf` in the L2 norms in this PR: https://github.com/OpenMined/PySyft/pull/7008. Indeed, I created an exploit that allowed a DS to publish entire arrays while only consuming the default budget for publishing noise. It seems like these checks were enough. On the overflow issue, however, I could reverse the min and max values, but couldn't abuse this in any way. I will add some warnings on the DS side for this issue https://github.com/OpenMined/PySyft/issues/6952, but at the moment in doesn't seem we need any check on the server. Let me know if you have any recommendations!
